### ISSUE TITLE
4CHR3 cannot turn on multiple channels.

### ISSUE THREAD

**Issue by Peetey** [bwp91/homebridge-ewelink#130]
**What issue do you have? Please be as thorough and explicit as possible.**
I have a 4CHR3 installed and working perfect through the eweLink app. All channels can be turned on and off independently without issues. However, I noticed that I cannot have more than 1 channel turned on through Homebridge. If I turn CH1 on then turn CH2 on, then it will turn off CH1. This applies to every channel, turning on a single channel will turn off all other active channels. The master switch CH0 still functions correctly. Homekit and HOOBS will still show the switches as on even though they're off.

Example:
1. Toggle CH1 (CH1 turns on)
2. Toggle CH2 (CH2 turns on and CH1 turns off)(Homekit and HOOBS still shows CH1 as on)
3. Toggle CH1 (CH2 turns off)(Homekit and HOOBS shows CH1 off but CH2 still on)
4. Toggle CH2 (Nothing happens)(Homekit and HOOBS now shows CH2 off)

The log below shows the steps I mentioned in the example above right after a fresh restart of HOOBS. It seems to be turning off other switches when I turn on one of them and turning everything off when 1 is turned off.
...

**Details of your setup.**
Do you use Homebridge or HOOBS?
HOOBS with a TadoAC plugin as the only other plugin.
...

Which version of Homebridge/HOOBS do you have?
3.2.9
...

Which version of this plugin (homebridge-ewelink) do you have?
3.10.0 but the issue has been there before I updated from 3.5
...

**Please paste any relevant logs below. It helps if you can turn `debug` and `debugReqRes` in the package settings for more thorough logging.**
> If you are posting an error then it is helpful for me to also see the previous few lines as this can show the cause of the error.

```
11/3/2020, 1:31:58 PM Loaded plugin 'homebridge-ewelink'
11/3/2020, 1:31:58 PM [11/2/2020, 11:31:58 PM] Registering platform 'homebridge-ewelink.eWeLink'
11/3/2020, 1:31:58 PM Loaded plugin 'homebridge-tado-ac'
11/3/2020, 1:31:58 PM [11/2/2020, 11:31:58 PM] Registering platform 'homebridge-tado-ac.TadoAC'
11/3/2020, 1:31:58 PM Loading 2 platforms...
11/3/2020, 1:31:58 PM [Tado AC] Initializing TadoAC platform...
11/3/2020, 1:31:58 PM [eWeLink] Initializing eWeLink platform...
11/3/2020, 1:31:58 PM [eWeLink] Plugin [v3.10.0] initialised. Syncing with eWeLink...
11/3/2020, 1:31:58 PM [eWeLink] Sending HTTP getHost() request. This text is yellow for clarity.
{
  "appid": "oeVkj2lYFGnJu5XUtWisfW4utiN4u9Mq",
  "country_code": "+66",
  "nonce": "4cbhfn5g",
  "ts": 1604385118,
  "version": 8
}
11/3/2020, 1:31:58 PM Bridge is running on port 51826.
11/3/2020, 1:31:58 PM [eWeLink] HTTP API host received [as-apia.coolkit.cc].
11/3/2020, 1:31:58 PM [eWeLink] Sending HTTP login() request. This text is yellow for clarity.
{
  "countryCode": "+66",
  "password": "**hidden**",
  "email": "**hidden**"
}
11/3/2020, 1:32:01 PM [eWeLink] Web socket host received [as-pconnect2.coolkit.cc].
11/3/2020, 1:32:01 PM [eWeLink] Sending WS login request. This text is yellow for clarity.
{
  "action": "userOnline",
  "apikey": "**hidden**",
  "appid": "oeVkj2lYFGnJu5XUtWisfW4utiN4u9Mq",
  "at": "**hidden**",
  "nonce": "ao3j1cqr",
  "sequence": "1604385121469",
  "ts": 1604385121,
  "userAgent": "app",
  "version": 8
}
11/3/2020, 1:32:04 PM [eWeLink] [5] eWeLink devices loaded from the Homebridge cache.
11/3/2020, 1:32:04 PM [eWeLink] [1] primary devices loaded from your eWeLink account.
11/3/2020, 1:32:04 PM [eWeLink]  → [4CH] initialised but LAN mode unavailable as device is shared (xpeetey+ewelink@gmail.com).
11/3/2020, 1:32:04 PM [eWeLink] eWeLink sync complete. Don't forget to ⭐️  this plugin on GitHub if you're finding it useful!
11/3/2020, 1:32:04 PM [eWeLink] Note: 'Request & Response Logging' is not advised for long-term use.
11/3/2020, 1:32:23 PM [eWeLink] [4CH SW1] Reverting to web socket as device isn't reachable via LAN mode.
11/3/2020, 1:32:23 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "on",
        "outlet": 0
      },
      {
        "switch": "off",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604385143651",
  "ts": 0,
  "userAgent": "app"
}
11/3/2020, 1:32:31 PM [eWeLink] [4CH SW2] Reverting to web socket as device isn't reachable via LAN mode.
11/3/2020, 1:32:31 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "off",
        "outlet": 0
      },
      {
        "switch": "on",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604385151216",
  "ts": 0,
  "userAgent": "app"
}
11/3/2020, 1:32:34 PM [eWeLink] [4CH SW1] Reverting to web socket as device isn't reachable via LAN mode.
11/3/2020, 1:32:34 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "off",
        "outlet": 0
      },
      {
        "switch": "off",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604385154943",
  "ts": 0,
  "userAgent": "app"
}
11/3/2020, 1:32:37 PM [eWeLink] [4CH SW2] Reverting to web socket as device isn't reachable via LAN mode.
11/3/2020, 1:32:37 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "off",
        "outlet": 0
      },
      {
        "switch": "off",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604385157529",
  "ts": 0,
  "userAgent": "app"
}
```

...

**Comment by bwp91** [comment]
Hi,
Are you using a different ewelink account for hoobs than for your ewelink app? like a shared account?

Edit:
Let me rephrase as I already see that the device is shared. My question is why are you using a shared device?

**Comment by Peetey** [comment]
I have my device shared with a secondary account so I could login to Ewelink with two phones. This is because the Ewelink app on iOS doesn't seem to be able to add new devices anymore so I had to use another android phone to setup switches. Ewelink seems to only allow one login at a time and I read somewhere on Reddit that sharing devices work so thats what I have right now.

**Comment by bwp91** [comment]
Having two ewelink accounts was needed until recently. Please try logging into your main ewelink account in hoobs.

And it shouldn’t log out out of your own phone app.

And i also believe this will fix your original issue!

**Comment by Peetey** [comment]
Thanks for the quick help.

I switched off sharing and used a single account for my phone and the plugin. The issue still persists. I made sure that interlocking is disabled in the Ewelink app as well. Here is a log for the same actions without sharing mode. I also attached a screenshot of the log in the Ewelink app.

![IMG_1696](https://user-images.githubusercontent.com/24828910/98071589-8ded5100-1e96-11eb-8833-96f9807ddd62.jpeg)

```
11/4/2020, 12:01:45 PM Loaded plugin 'homebridge-ewelink'
11/4/2020, 12:01:45 PM [11/3/2020, 10:01:45 PM] Registering platform 'homebridge-ewelink.eWeLink'
11/4/2020, 12:01:45 PM Loaded plugin 'homebridge-tado-ac'
11/4/2020, 12:01:45 PM [11/3/2020, 10:01:45 PM] Registering platform 'homebridge-tado-ac.TadoAC'
11/4/2020, 12:01:45 PM Loading 2 platforms...
11/4/2020, 12:01:45 PM [Tado AC] Initializing TadoAC platform...
11/4/2020, 12:01:45 PM [eWeLink] Initializing eWeLink platform...
11/4/2020, 12:01:45 PM [eWeLink] Plugin [v3.10.0] initialised. Syncing with eWeLink...
11/4/2020, 12:01:45 PM [eWeLink] Sending HTTP getHost() request. This text is yellow for clarity.
{
  "appid": "oeVkj2lYFGnJu5XUtWisfW4utiN4u9Mq",
  "country_code": "+66",
  "nonce": "l342i16q",
  "ts": 1604466105,
  "version": 8
}
11/4/2020, 12:01:45 PM Bridge is running on port 51826.
11/4/2020, 12:01:46 PM [eWeLink] Sending HTTP login() request. This text is yellow for clarity.
{
  "countryCode": "+66",
  "password": "**hidden**",
  "email": "**hidden**"
}
11/4/2020, 12:01:48 PM [eWeLink] Sending WS login request. This text is yellow for clarity.
{
  "action": "userOnline",
  "apikey": "**hidden**",
  "appid": "oeVkj2lYFGnJu5XUtWisfW4utiN4u9Mq",
  "at": "**hidden**",
  "nonce": "no69tg7s",
  "sequence": "1604466108992",
  "ts": 1604466108,
  "userAgent": "app",
  "version": 8
}
11/4/2020, 12:01:51 PM [eWeLink] [5] eWeLink devices loaded from the Homebridge cache.
11/4/2020, 12:01:51 PM [eWeLink] [1] primary devices loaded from your eWeLink account.
11/4/2020, 12:01:51 PM [eWeLink]  → [4CH] initialised but LAN mode unavailable as device is unreachable.
11/4/2020, 12:01:51 PM [eWeLink] eWeLink sync complete. Don't forget to ⭐️  this plugin on GitHub if you're finding it useful!
11/4/2020, 12:01:51 PM [eWeLink] Note: 'Request & Response Logging' is not advised for long-term use.
11/4/2020, 12:02:23 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "on",
        "outlet": 0
      },
      {
        "switch": "off",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604466143426",
  "ts": 0,
  "userAgent": "app"
}
11/4/2020, 12:02:23 PM [eWeLink] WS message received.
{
  "deviceid": "**hidden**",
  "params": {
    "online": true,
    "updateSource": "WS"
  }
}
11/4/2020, 12:02:24 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "off",
        "outlet": 0
      },
      {
        "switch": "on",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604466144546",
  "ts": 0,
  "userAgent": "app"
}
11/4/2020, 12:02:24 PM [eWeLink] WS message received.
{
  "deviceid": "**hidden**",
  "params": {
    "online": true,
    "updateSource": "WS"
  }
}
11/4/2020, 12:02:29 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "off",
        "outlet": 0
      },
      {
        "switch": "off",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604466149817",
  "ts": 0,
  "userAgent": "app"
}
11/4/2020, 12:02:29 PM [eWeLink] WS message received.
{
  "deviceid": "**hidden**",
  "params": {
    "online": true,
    "updateSource": "WS"
  }
}
11/4/2020, 12:02:30 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "off",
        "outlet": 0
      },
      {
        "switch": "off",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604466150380",
  "ts": 0,
  "userAgent": "app"
}
11/4/2020, 12:02:30 PM [eWeLink] WS message received.
{
  "deviceid": "**hidden**",
  "params": {
    "online": true,
    "updateSource": "WS"
  }
}
```

**Comment by bwp91** [comment]
This line here

```
11/4/2020, 12:01:51 PM [eWeLink]  → [4CH] initialised but LAN mode unavailable as device is unreachable.
```

makes me question of your device has a stable internet connection? Is the device on your home network?

The time differences between the logs and the screenshot you send are about 5 minutes apart?

In general, the issue you are having is that the plugin is not receiving "ok" responses from eWeLink when you send a device update, and so the plugin doesn't update what it thinks the state of the devices are.

**Comment by Peetey** [comment]
The 4CH has strong internet connection but is on another Wifi access point and thus SSID. I have multiple APs throughout my house all linked through LAN. The response time is instant so I don't think its a signal stregnth issue. My phone and computer has been connected to the same AP and SSID the HOOBS is on this whole time.

The log screenshot is me performing the same actions a few minutes after I pasted the HOOBS log so the time doesn't match up. Sorry I did not make that clear.

Maybe there is an issue with getting an issue with getting an "ok" response from the 4CH because its on another AP? My phone controls it just fine with the Ewelink app regardless of which AP its connected to though.

**Comment by bwp91** [comment]
> Maybe there is an issue with getting an issue with getting an "ok" response from the 4CH because its on another AP? My phone controls it just fine with the Ewelink app regardless of which AP its connected to though.

This shouldn't be the case. This is the reason why the plugin can't detect your device through LAN mode, but, the 'ok' updates from eWeLink are sent straight to Homebridge, not via the device.

The plugin is receiving some sort of response from eWeLink each time you try to control the device:

```
11/4/2020, 12:02:30 PM [eWeLink] WS message received.
{
  "deviceid": "**hidden**",
  "params": {
    "online": true,
    "updateSource": "WS"
  }
}
```

It is just not very informative, and not in the format of a usual standard response. This is what is puzzling me.

Is there the option of giving this device a static IP over your local network?

**Comment by Peetey** [comment]
I currently can't set a static IP for it yet. For some reason I can't access the control panel for that specific AP when I put in it's IP. I'll try moving the 4CH to another AP for now and see if that changes anything.

**Comment by bwp91** [comment]
At the moment your device is being controlled over the cloud, and for some reason it's not receiving updates back from eWeLink.

If you are able to give the device a static IP then you can use the `ipOverride` configuration option which will hopefully let you control the device over LAN mode, which is a great improvement over the cloud and should get it working properly again.

Let me know if you are able to give the device a static IP and I can help you with the setup for the plugin.

**Comment by bwp91** [comment]
Please update to v3.11.1 you might find an improvement :)

**Comment by Peetey** [comment]
Really appreciate the help you've been giving me so far. Some weird things to report with the latest update though! I haven't got around to changing the network setup yet so we can still try a static IP once I figure out my AP's control panel. Right after updating I tried out the usual sequence and the both switches stayed on. However the third and fourth switch started malfunctioning where toggling SW1 will also toggle SW3. I didn't have much time to test it out and enable debugging log then so I left it and came back today.

Fortunately the weird stuff stopped but it's back to how it was before the update. Here's the log for it:

```
11/7/2020, 4:52:07 PM [eWeLink] Initializing eWeLink platform...
11/7/2020, 4:52:07 PM [eWeLink] Plugin [v3.11.1] initialised. Syncing with eWeLink...
11/7/2020, 4:52:07 PM [eWeLink] Sending HTTP getHost() request. This text is yellow for clarity.
{
  "appid": "oeVkj2lYFGnJu5XUtWisfW4utiN4u9Mq",
  "country_code": "+66",
  "nonce": "forbzkdk",
  "ts": 1604742727,
  "version": 8
}
11/7/2020, 4:52:07 PM Bridge is running on port 51826.
11/7/2020, 4:52:07 PM [eWeLink] HTTP API host received [as-apia.coolkit.cc].
11/7/2020, 4:52:07 PM [eWeLink] Sending HTTP login() request. This text is yellow for clarity.
{
  "countryCode": "+66",
  "password": "**hidden**",
  "email": "**hidden**"
}
11/7/2020, 4:52:10 PM [eWeLink] Web socket host received [as-pconnect1.coolkit.cc].
11/7/2020, 4:52:11 PM [eWeLink] Sending WS login request. This text is yellow for clarity.
{
  "action": "userOnline",
  "apikey": "**hidden**",
  "appid": "oeVkj2lYFGnJu5XUtWisfW4utiN4u9Mq",
  "at": "**hidden**",
  "nonce": "2j51h5oh",
  "sequence": "1604742731009",
  "ts": 1604742731,
  "userAgent": "app",
  "version": 8
}
11/7/2020, 4:52:13 PM [eWeLink] [5] eWeLink devices loaded from the Homebridge cache.
11/7/2020, 4:52:13 PM [eWeLink] [1] primary devices loaded from your eWeLink account.
11/7/2020, 4:52:13 PM [eWeLink]  → [4CH] initialised but LAN mode unavailable as device is unreachable.
11/7/2020, 4:52:13 PM [eWeLink] eWeLink sync complete. Don't forget to ⭐️  this plugin on GitHub if you're finding it useful!
11/7/2020, 4:52:13 PM [eWeLink] Note: 'Request & Response Logging' is not advised for long-term use.
11/7/2020, 4:52:20 PM [eWeLink] [4CH SW1] Reverting to web socket as device isn't reachable via LAN mode.
11/7/2020, 4:52:20 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "on",
        "outlet": 0
      },
      {
        "switch": "off",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604742740418",
  "ts": 0,
  "userAgent": "app"
}
11/7/2020, 4:52:20 PM [eWeLink] WS message received.
{
  "deviceid": "**hidden**",
  "params": {
    "online": true,
    "updateSource": "WS"
  }
}
11/7/2020, 4:52:20 PM [eWeLink] [4CH] WS update received and will be refreshed.
11/7/2020, 4:52:22 PM [eWeLink] [4CH SW2] Reverting to web socket as device isn't reachable via LAN mode.
11/7/2020, 4:52:22 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "off",
        "outlet": 0
      },
      {
        "switch": "on",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604742742699",
  "ts": 0,
  "userAgent": "app"
}
11/7/2020, 4:52:22 PM [eWeLink] WS message received.
{
  "deviceid": "**hidden**",
  "params": {
    "online": true,
    "updateSource": "WS"
  }
}
11/7/2020, 4:52:22 PM [eWeLink] [4CH] WS update received and will be refreshed.
11/7/2020, 4:52:33 PM [eWeLink] [4CH SW1] Reverting to web socket as device isn't reachable via LAN mode.
11/7/2020, 4:52:33 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "off",
        "outlet": 0
      },
      {
        "switch": "off",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604742753529",
  "ts": 0,
  "userAgent": "app"
}
11/7/2020, 4:52:33 PM [eWeLink] WS message received.
{
  "deviceid": "**hidden**",
  "params": {
    "online": true,
    "updateSource": "WS"
  }
}
11/7/2020, 4:52:33 PM [eWeLink] [4CH] WS update received and will be refreshed.
11/7/2020, 4:52:36 PM [eWeLink] [4CH SW2] Reverting to web socket as device isn't reachable via LAN mode.
11/7/2020, 4:52:36 PM [eWeLink] WS message sent. This text is yellow for clarity.
{
  "apikey": "**hidden**",
  "deviceid": "**hidden**",
  "params": {
    "switches": [
      {
        "switch": "off",
        "outlet": 0
      },
      {
        "switch": "off",
        "outlet": 1
      },
      {
        "switch": "off",
        "outlet": 2
      },
      {
        "switch": "off",
        "outlet": 3
      }
    ]
  },
  "action": "update",
  "sequence": "1604742756092",
  "ts": 0,
  "userAgent": "app"
}
11/7/2020, 4:52:36 PM [eWeLink] WS message received.
{
  "deviceid": "**hidden**",
  "params": {
    "online": true,
    "updateSource": "WS"
  }
}
11/7/2020, 4:52:36 PM [eWeLink] [4CH] WS update received and will be refreshed
```

**Comment by bwp91** [comment]
Would it be possible to message you on the [Homebridge Discord](https://discord.gg/7X36mdw)?

**Comment by fizulnizam** [comment]
Hi there guys, I am having the same issue as well..

Edit: I only use one account on Ewelink and on the HOOBS plugin of yours yet I still face the same issue.

**Comment by bwp91** [comment]
@fizulnizam 
Please try updating to the beta version using the instructions here:

https://github.com/bwp91/homebridge-ewelink/wiki/Beta-Version#hoobs

**Comment by TassiRob** [comment]
Came here as new homebridge user, trying to figure out how to code my config correctly to use a 4Ch-RF Smart Switch. Forgive me if this is a bumb question.

I note in the device settings there is an interlock feature that is designed to produce this behavior. Has it been enabled within the e-Welink app by the end user or is it being enabled by the plugin?

**Comment by bwp91** [comment]
Hi @TassiRob 
I'm slightly confused at 

> I note in the device settings there is an interlock feature that is designed to produce this behavior

As I have not implemented any sort of interlock feature in the plugin? Which setting are you talking about?

**Comment by TassiRob** [comment]
Within the eWeLink App one of the deivce settings for the 4 channel switch that I have is an "Interlock" option that is described as producing this precise behavior. 
Was wondering if it had been enabled within the eWeLink app by mistake or if, in the implementation of the plugin, this "feature" is getting enabled.

**Comment by bwp91** [comment]
The plugin does not turn on/off the interlock feature of any device, if that is what you are asking.

If you have interlocking turned on for the device, the plugin should mimic the behaviour of this from the updates that it gets from eWeLink.

I hope I am answering your questions :)

**Comment by TassiRob** [comment]
I am not being clear. My question was speculation about what was causing the problem for the original poster. 
'Could they have turned it on in their eWeLink App by mistake?'
I haven't got any problems other than learning how to use homebridge in general. Have now got my 4 channel switch working just as I want it to. 
Thanks for all your work and sorry for not being clearer earlier.

**Comment by fizulnizam** [comment]
@TassiRob its okay you were just trying to help.. Just for your information, I have checked whether or not interlock was enabled in the ewelink app but no it was not. I tried enabling and disabling the interlock feature but there wasn't difference to how the switch behaved. Am trying to install the beta version now, will update the progress soon.

**Comment by bwp91** [comment]
Ahh @TassiRob now I understand! I thought you were having an issue yourself with the interlock getting turned on somehow! I appreciate your suggestion here 👍

**Comment by fizulnizam** [comment]
Hey man @bwp91, I just tried the 3.12.0-5 beta and it worked wonderfully! Theres a small (maybe around half a second) lag between the switches turning on but I am soo grateful for the fix! Thank you so much Ben! Hopefully it will work for @Peetey as well.

**Comment by bwp91** [comment]
@fizulnizam the slight delay I have had to introduce to fix a separate issue. It’s a delay of a random time between 0 and 1 second, so hopefully it’s not too irritating! 
I’m glad it’s working for you. you are fine to stay on the beta until i release the update which will appear as an update once it’s released. 
now there’s a tongue twister!

**Comment by bwp91** [comment]
v3.12.0 seems to resolve this issue :)

**Comment by bwp91** [comment]
Gonna get this issue closed as @fizulnizam you've said its's working okay and @Peetey we got this sorted on the discord.
Any further issues do let me know!
